src/contracts/core/DelegationManager.sol

```diff
@@ -45,107 +45,155 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
-
-    /// @dev Emitted when a low-level call to `delegationTerms.onDelegationWithdrawn` fails, returning `returnData`
-    event OnDelegationWithdrawnCallFailure(IDelegationTerms indexed delegationTerms, bytes32 returnData);
-
-    /// @dev Emitted when an entity registers itself as an operator in the DelegationManager
-    event RegisterAsOperator(address indexed operator, IDelegationTerms indexed delegationTerms);
-
     function initialize(address initialOwner, IPauserRegistry _pauserRegistry, uint256 initialPausedStatus)
         external
         initializer
     {
         _initializePauser(_pauserRegistry, initialPausedStatus);
-        DOMAIN_SEPARATOR = keccak256(abi.encode(DOMAIN_TYPEHASH, keccak256(bytes("EigenLayer")), ORIGINAL_CHAIN_ID, address(this)));
+        _DOMAIN_SEPARATOR = _calculateDomainSeparator();
         _transferOwnership(initialOwner);
     }
 
     // EXTERNAL FUNCTIONS
     /**
-     * @notice This will be called by an operator to register itself as an operator that stakers can choose to delegate to.
-     * @param dt is the `DelegationTerms` contract that the operator has for those who delegate to them.
-     * @dev An operator can set `dt` equal to their own address (or another EOA address), in the event that they want to split payments
-     * in a more 'trustful' manner.
-     * @dev In the present design, once set, there is no way for an operator to ever modify the address of their DelegationTerms contract.
+     * @notice Registers the `msg.sender` as an operator in EigenLayer, that stakers can choose to delegate to.
+     * @param registeringOperatorDetails is the `OperatorDetails` for the operator.
+     * @param metadataURI is a URI for the operator's metadata, i.e. a link providing more details on the operator.
+     * @dev Note that once an operator is registered, they cannot 'deregister' as an operator, and they will forever be considered "delegated to themself".
+     * @dev This function will revert if the caller attempts to set their `earningsReceiver` to address(0).
+     * @dev Note that the `metadataURI` is *never stored in storage* and is instead purely emitted in an `OperatorMetadataURIUpdated` event
      */
-    function registerAsOperator(IDelegationTerms dt) external {
+    function registerAsOperator(OperatorDetails calldata registeringOperatorDetails, string calldata metadataURI) external {
         require(
-            address(delegationTerms[msg.sender]) == address(0),
+            _operatorDetails[msg.sender].earningsReceiver == address(0),
             "DelegationManager.registerAsOperator: operator has already registered"
         );
-        // store the address of the delegation contract that the operator is providing.
-        delegationTerms[msg.sender] = dt;
-        _delegate(msg.sender, msg.sender);
-        emit RegisterAsOperator(msg.sender, dt);
+        _setOperatorDetails(msg.sender, registeringOperatorDetails);
```

Comment:
should this be called `_modifyOperatorDetails` or the external function be called `_setOperatorDetails` for consistency?

Reply:
it's used both in `registerAsOperator` and `modifyOperatorDetails`, so I think giving the internal function a different name is more appropriate (since it _isn't_ just an alias for the single external function)